### ISSUE TITLE
[css-ruby] Parent doesn't contain children

### ISSUE THREAD

**Issue by cjbarth** [w3c/csswg-drafts#3530]
Spec: https://drafts.csswg.org/css-ruby-1/#line-height

[This bug report](https://bugzilla.mozilla.org/show_bug.cgi?id=1513904) with Firefox and the related discussion illustrate a potential implementation problem when dealing with <ruby> text and trying to position it when it overflows its parent container, as it is currently implement to do and which the spec seems to call for. If a parent container for <ruby> elements behaves differently when sizing itself compared to a container with only normal text elements, styling becomes very problematic as the attachment to that Firefox bug illustrates.

**Comment by fantasai** [comment]
@cjbarth Yeah, so the reasoning behind the current spec is to avoid increasing the line-spacing for ruby in the general case: typographers really don't like it when that happens. The side-effect as you noticed, is that the ruby leaks halfway outside the block box.

We can add a switch to say "line box must contain 100% of the ruby", and that's probably appropriate in some cases; but it does mean that lines with ruby and lines without ruby will be spaced differently. I'm not sure whether that's what you want, or whether you want to always reserve space as if ruby was there, per #3351. Would you mind clarifying?

**Comment by cjbarth** [comment]
I feel like my concern is the fact that highlighting can't be applied to containers with `ruby` in them. It does seem like it would be an appropriate solution to have some option to reserve space as if ruby was there. I see that the draft has been updated to mention about "CSS Line Layout Module Level 3" since this bug was opened.

Honestly, I kinda expect that lines with ruby and lines without ruby would be spaced differently, as they are in Word for example. However, typographically speaking, I can see how it would be very advantageous to be able to set a property that would force all lines to be sized as if they had ruby. That would allow a guarantee fit if a line ended up with ruby (like Japanese), but would also allow the document to be styled even if the ruby wasn't present and still have it look consistent and uniform. It would also allow for turning on and off ruby by changing its `visibility` or `display` without causing a reflow.
